polyfill/lib/calendar.mjs

```diff
@@ -1013,22 +995,30 @@ const nonIsoHelperBase = {
   // All built-in calendars except Chinese/Dangi and Hebrew use an era
   hasEra: true,
   monthDayFromFields(fields, overflow, cache) {
-    let { year, month, monthCode, day, era, eraYear } = fields;
+    let { monthCode, day } = fields;
     if (monthCode === undefined) {
+      let { year, era, eraYear } = fields;
       if (year === undefined && (era === undefined || eraYear === undefined)) {
         throw new TypeError('`monthCode`, `year`, or `era` and `eraYear` is required');
```

Comment:
While we're here, maybe clarify this message a bit?  Like this?

when `monthCode` is omitted, `year` (or `era` and `eraYear`) and `month` are required